orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/CapabilitiesController.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.controllers;
+
+import com.netflix.spinnaker.config.MonitoredDeployConfigurationProperties;
+import com.netflix.spinnaker.orca.model.DeploymentMonitorDefinition;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/** Controller intended for querying various capabilities of orca */
+@RestController
+public class CapabilitiesController {
+  private MonitoredDeployConfigurationProperties monitoredDeployConfigurationProperties;
+
+  public CapabilitiesController(
+      Optional<MonitoredDeployConfigurationProperties> monitoredDeployConfigurationProperties) {
+    this.monitoredDeployConfigurationProperties =
+        monitoredDeployConfigurationProperties.orElse(null);
+  }
+
+  @GetMapping("/capabilities/deploymentMonitors")
+  public List<DeploymentMonitorDefinition> getDeploymentMonitors() {
+    ArrayList<DeploymentMonitorDefinition> result = new ArrayList<>();
+
+    if (monitoredDeployConfigurationProperties != null) {
+      monitoredDeployConfigurationProperties
```

Comment:
could _streamline_ this with streams:

```
if (monitoredDeployConfigurationProperties == null) {
    return Collections.emptyList();
}

return monitoredDeployConfigurationProperties
    .getDeploymentMonitors()
    .stream()
    .map(dm -> result.add(new DeploymentMonitorDefinition(dm)))
    .collect(Collectors.toList());
```

Reply:
thanks, i like it (even though i am not a fan of  java streams)